src/games/supported/YarsRevenge.cpp

```diff
@@ -134,3 +138,41 @@ ActionVect YarsRevengeSettings::getStartingActions() {
     startingActions.push_back(PLAYER_A_FIRE);
     return startingActions;
 }
+
+// returns a list of mode that the game can be played in
+ModeVect YarsRevengeSettings::getAvailableModes(){
+    ModeVect modes;
+    modes.push_back(0);
+    modes.push_back(0x20);
+    modes.push_back(0x40);
+    modes.push_back(0x60);
+    return modes;
+}
+
+// set the mode of the game
+// the given mode must be one returned by the previous function
+void YarsRevengeSettings::setMode(game_mode_t m, System &system, StellaEnvironment& environment){
+    if(m == 0 || m == 0x20 || m == 0x40 || m == 0x60 ){
```

Comment:
The difference is the speed of the auto guided missile : in mode 0, it takes 12seconds to reach the glittering area, vs only 10 seconds in mode 0x20. (yes the difference is tiny)
However, I noticed that the default mode of this game is 0x20, I'll init m_mode accordingly